hnn_core/network.py

```diff
@@ -35,27 +37,54 @@ def read_spikes(fname, gid_dict=None):
     spike_times = []
     spike_gids = []
     spike_types = []
+    spike_tstart = []
+    spike_tstop = []
     for file in sorted(glob(fname)):
         spike_trial = np.loadtxt(file, dtype=str)
         spike_times += [list(spike_trial[:, 0].astype(float))]
         spike_gids += [list(spike_trial[:, 1].astype(int))]
 
         # Note that legacy HNN 'spk.txt' files don't contain a 3rd column for
-        # spike type. If reading a legacy version, validate that a gid_dict is
-        # provided.
-        if spike_trial.shape[1] == 3:
+        # spike type, or a 4th and 5th column for tstart and tstop. If reading
+        # a legacy version, validate that gid_dict, tstart, and tstop provided.
+        if spike_trial.shape[1] >= 3:
             spike_types += [list(spike_trial[:, 2].astype(str))]
-        else:
+        elif spike_trial.shape[1] == 2:
             if gid_dict is None:
                 raise ValueError("gid_dict must be provided if spike types "
                                  "are unspecified in the file %s" % (file,))
             spike_types += [[]]
 
-    spikes = Spikes(times=spike_times, gids=spike_gids, types=spike_types)
+        if spike_trial.shape[1] == 5:
+            spike_tstart += np.unique(spike_trial[:, 3].astype(float)).tolist()
+            spike_tstop += np.unique(spike_trial[:, 4].astype(float)).tolist()
+        else:
+            if tstart is None or tstop is None:
+                raise ValueError("tstart and tstop must be provided if values "
+                                 "are unspecified in the file %s" % (file,))
+
+    if len(np.unique(spike_tstart)) > 1 or len(np.unique(spike_tstop)) > 1:
+        raise ValueError("tstart and tstop must match across files.")
+
+    elif len(np.unique(spike_tstart)) == 1 and len(
+            np.unique(spike_tstop)) == 1:
+        spike_tstart = spike_tstart[0]
+        spike_tstop = spike_tstop[0]
+    else:
+        spike_tstart = np.min(sum(spike_times, []))
+        spike_tstop = np.max(sum(spike_times, []))
```

Comment:
I would say we shouldn't guess if the values are not given. It would give incorrect results, no?

Also, given [L62-64](https://github.com/jonescompneurolab/hnn-core/pull/155/files#diff-3e7f4ea1f282c3abd4802eef071e9a2aR62-R64), is there any chance of hitting this `else` condition?

Reply:
Definitely a brain fart there, I agree this won't be reached. I'll convert the preceding `elif` clause to `else`.